fearless/Modules/Staking/Analytics/Common/AnalyticsViewModelFactoryBase.swift

```diff
@@ -0,0 +1,118 @@
+import BigInt
+import SoraFoundation
+
+protocol AnalyticsViewModelItem: Dated {
+    var timestamp: Int64 { get }
+    var amount: BigUInt { get }
+}
+
+class AnalyticsViewModelFactoryBase<T: AnalyticsViewModelItem> {
+    let chain: Chain
+    let balanceViewModelFactory: BalanceViewModelFactoryProtocol
+
+    init(
+        chain: Chain,
+        balanceViewModelFactory: BalanceViewModelFactoryProtocol
+    ) {
+        self.chain = chain
+        self.balanceViewModelFactory = balanceViewModelFactory
+    }
+
+    func createViewModel(
```

Comment:
Please, break up this function. At least we can move calculation of `AnalyticsRewardsViewModel` part to a separate functions.